pkg/kapp/app/recorded_app.go

```diff
@@ -144,14 +143,37 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string, isDiffRun bool) (
 	if err != nil {
 		return false, err
 	}
-
 	if foundNonMigratedApp {
 		if a.isMigrationEnabled() {
 			return false, a.migrate(app, labels, a.fqName())
 		}
 		return false, a.updateApp(app, labels)
 	}
 
+	if prevAppName == "" {
+		return true, a.create(labels, isDiffRun)
+	}
+
+	app, foundMigratedPrevApp, err := a.find(prevAppName + AppSuffix)
```

Comment:
`prevAppName` would be empty here was we are returning above otherwise 🤔

Reply:
At this stage, `app` is not present and `prevAppName` is _not_ empty.

Reply:
Oh yes, looks like I misread the condition above. Makes sense!